### PR TITLE
Implements Sanctions

### PR DESCRIPTION
**Pull request by Squigger**
## About The Pull Request

Sanctions can now be issued against xenos for their unlawful occupation of the colony. This does nothing.

## Why It's Good For The Game

image

## Changelog
:cl:
add: Sanctions can now be issued.
/:cl:

### CODE REVIEW

**Comment by Lefunny** [comment]
Based and necessary speedmerge please

**Comment by AGasCanister** [comment]
image

**Comment by revelation-8468** [comment]
image

**Comment by lKiseki** [comment]
image

**Comment by Passe2142** [comment]
image

**Comment by novaepee** [comment]
image

**Comment by MiniMeatwad** [comment]
can you also issue multiple sanctions to really rub it in that the xenos are gonna pay for this?

**Comment by jroinc** [comment]
Do ayys hear this tho?

**Comment by Lumipharon** [comment]
Can central command send the outpost Hopes and Prayers?

**Comment by Squigger** [comment]
> Do ayys hear this tho?

Should now be visible via a QM announcement to xenos.

> can you also issue multiple sanctions to really rub it in that the xenos are gonna pay for this?

There is no cooldown or disabling of this.

> Can central command send the outpost Hopes and Prayers?

image

**Comment by Kuro020** [comment]
"xeno mains dont go to purgatory queen, they go straight to hell" - Captain probably

**Comment by Luxyboo** [comment]
image

**Comment by BonkaiB** [comment]
image

**Comment by David-Stormwell** [comment]
speed merhe

**Comment by yyzsong** [comment]
image

**Comment by Ragecrabbo** [comment]
https://user-images.githubusercontent.com/65325586/155618068-43224dc4-6fa7-4070-875c-940e223685dd.png

**Comment by novaepee** [comment]
image

File: code/game/objects/machinery/computer/communications.dm [hunk 1]
```diff
@@ -119,6 +119,15 @@
 				log_game("[key_name(usr)] has just sent a command announcement.")
 				cooldown_message = world.time
 
+		if("sanction")
+			if(authenticated == 2)
+				priority_announce("The TerraGov Marine Corps have issued formal sanctions against the xenos for their unjustified invasion of the colony.", type = ANNOUNCEMENT_PRIORITY)
+				for(var/i in (GLOB.xeno_mob_list + GLOB.observer_list))
+					var/mob/M = i
```
Comment (Kapu1178) [inline code/game/objects/machinery/computer/communications.dm:126]:
```suggestion
				for(var/mob/I as anything in (GLOB.xeno_mob_list + GLOB.observer_list))
```

**Comment by Losenis** [comment]
New sanction, psypoint embargo. Makes marines not psydrainable.

**Comment by Ragecrabbo** [comment]
> New sanction, psypoint embargo. Makes marines not psydrainable.

The TerraGov Marine Corps rely too much on xenomorph mining industry for this to be possible. Any more sanctions and xenos will start destroying miners rather than just disabling them.

File: code/game/objects/machinery/computer/communications.dm (hunk 1 above)
Comment (Squigger) [reply code/game/objects/machinery/computer/communications.dm:126]: > image It's the same thing except in 1 less line. `as anything` skips the implicit typecheck when looping through a list and assigning to a typed var.

image

**Pull request closed without merging**